### DISCUSSION TITLE
Supported Extension

### DISCUSSION THREAD

**Discussion by Surventurer** [iina/iina#4855] [Q&A]
Tell me the all the extension it includes.

**Comment by low-batt** [comment]
A plugin system for IINA is being implemented. It is available in IINA 1.3.4, but not visible by default as it is still undergoing development and has not been officially released. The documentation for it can be found [here](https://docs.iina.io/index.html).

To enable the plugin system open a [Terminal](https://support.apple.com/guide/terminal/welcome/mac) window and cut-n-paste this command into the window and press return:
```bash
defaults write com.colliderli.iina iinaEnablePluginSystem true
```

Then:
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel look for a `Plugins` tab

If you see the new tab then the plugin system has been enabled.

To disable the plugin system merely repeat the above command setting the value to `false`.

For audio/video playback support IINA is using a library from the [mpv](https://mpv.io/) project. As a result the techniques discussed in [Using mpv from other programs or scripts](https://mpv.io/manual/stable/#using-mpv-from-other-programs-or-scripts) are also available such as [Lua Scripting](https://mpv.io/manual/stable/#lua-scripting).

To enable IINA to load scripts from mpv's config directory follow these instructions:
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Advanced`
- Slide the `Enable advanced settings` toggle button to be on (blue)
- Click on the checkbox `Use config directory` 
- **Restart IINA** to activate these settings

Many `mpv` scripting features will work, but not all, such as features that manipulate mpv's UI.

**Comment by lhc70000** [comment]
Thanks @low-batt! But seems that you provided the wrong documentation link. The correct location should be https://docs.iina.io/index.html.

**Comment by low-batt** [comment]
Thanks for catching my mistake. I've updated my post with the correct link.
